### PR TITLE
Externalize JavaScript: some easy files

### PR DESCRIPTION
**Pull request by orangejenny**
```
$ ./scripts/codechecks/hqDefine.sh

90%     (717/798) of HTML files are free of inline scripts
52%     (225/437) of JS files use hqDefine
11%     (45/437) of JS files specify their dependencies
```

90% of the way there...

@czue / @jmtroth0

### CODE REVIEW

File: corehq/apps/hqpillow_retry/static/hqpillow_retry/js/single.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        if (action === 'delete') {
+            var r=confirm("Are you sure you want to delete this error log?");
+            if (!r) {
+                return;
+            }
+        }
+        $("input[name=action]").val(action);
+        $("#actionform").submit();
+    }
+    $(function(){
```
Comment (czue) [inline corehq/apps/hqpillow_retry/static/hqpillow_retry/js/single.js]: probably just revealing my ignorance but how does this get called? does just passing it through `hqDefine` resolve the `onready` thing at the proper time?

File: corehq/apps/export/templates/export/partial/daily_export_move_notifier.html [hunk 2]
```diff
@@ -21,10 +21,3 @@ <h4 class="modal-title">
         </div>
     </div>
 </div>
-{% addtoblock js-inline %}
-<script>
-    $(window).on('load', function(){
-        $('#dailySavedExportNotifier').modal('show');
```
Comment (czue) [inline corehq/apps/export/templates/export/partial/daily_export_move_notifier.html]: this just gets handled by magic bootstrap js?

File: corehq/apps/userreports/static/userreports/js/data_source_evaluator.js [hunk 3]
```diff
@@ -43,6 +43,15 @@ hqDefine('userreports/js/data_source_evaluator', function () {
                 });
             }
         };
+
+        return self;
     };
-    return {DataSourceModel: DataSourceModel};
+
+    $(function () {
```
Comment (czue) [inline corehq/apps/userreports/static/userreports/js/data_source_evaluator.js]: (same question)

**Review by czue — APPROVED** [review]
I had a few questions just to wrap my head around things, but assuming you have sensible answers this looks great.

File: corehq/apps/export/templates/export/partial/daily_export_move_notifier.html (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/export/templates/export/partial/daily_export_move_notifier.html]: Kind of - modals actually default to being open. Typically we add the `fade` class, which hides them, so then you need js to open them. So removing `fade` removed the need for js.

File: corehq/apps/hqpillow_retry/static/hqpillow_retry/js/single.js (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/hqpillow_retry/static/hqpillow_retry/js/single.js]: yep, `hqDefine` takes care of it

Comment (czue) [reply corehq/apps/hqpillow_retry/static/hqpillow_retry/js/single.js]: neat. :tophat:

**Review by jmtroth0 — APPROVED** [review]

**Pull request merged**